code/Python/dbd.py

```diff
@@ -250,7 +250,10 @@ class definitions(Grammar):
     def grammar_elem_init(self, sessiondata):
 
         def flatten(lis, rec_depth=0):
-            from collections import Iterable
+            try:
+                from collections.abc import Iterable
+            except ImportError:
+                from collections import Iterable  # Python <3.3 compatibility
```

Comment:
I vote we just make this use `collections.abc` and ditch compatibility for python < 3.3 -- it's a decade old and barely even falls under the umbrella of "modern python". If someone really needs support for python _that_ old, they can always tweak the module themselves, or use an older version. I think 3.6 (out of support but reasonably modern) or 3.7 (supported, though not for long) would be a reasonable minimum version.

Hell, I can't even *install* 3.3 on my macbook.

[also, we should really move the import up to the top of the file, but that's neither here nor there]

Reply:
re the latter: I'm not a python person, so don't expect any compliance to general python rules ;) This is also a generally horrible API and everything since it is just a quickly thrown together thing that nobody has ever used in production. That snippet comes from https://stackoverflow.com/a/2158532 and my "be as local as possible"-brain made me import as locally as possible.

----------------------

The reason this is giving python <3.3 compatibility is since I think it even tries to be 2.7 compatible:

```
% vermin --backport argparse -vvv . | grep '!2'
!2, 3.10     ~/Documents/Blizzard/WoWDBDefs/code/Python/dbd.py
  L246: union types as `X | Y` requires !2, 3.10
```
which is a false-positive from what I can see: that's not type union, it is overloaded bitwise-or.

Why do this absurdity in the year of the Lord 2022? IDA supports 2.7 and I did not want to impose restrictions myself for using it there.

The minimum required version according to vermin (after removing that probably-false-positive) is 2.7 and 3.1. Bumping to 3.7 just for sake of not having this import-fallback seems.. meh.

Reply:
(And yes, I realise that #100 bumps that to 
```
2.7, 3.0     /Users/berndloerwald/Documents/Blizzard/WoWDBDefs/code/Python/dbd.py
~2, ~3       /Users/berndloerwald/Documents/Blizzard/WoWDBDefs/code/Python/dbd_grammar.py
!2, 3.6      /Users/berndloerwald/Documents/Blizzard/WoWDBDefs/code/Python/dbd_to_mysql.py
2.7, 3.1     /Users/berndloerwald/Documents/Blizzard/WoWDBDefs/code/Python/dbd_to_wiki.py
2.7, 3.1     /Users/berndloerwald/Documents/Blizzard/WoWDBDefs/code/Python/dbd_to_wmvxml.py
!2, 3.8      /Users/berndloerwald/Documents/Blizzard/WoWDBDefs/code/Python/dbdwrapper.py
2.7, 3.1     /Users/berndloerwald/Documents/Blizzard/WoWDBDefs/code/Python/dump_relations.py
```
, I see why one wants to do that, and I'd probably do it myself in other code. I think it is perfectly fine if it is just in the wrapper and to-MySQL, since the IDA scripts don't use that, so the overall requirement isn't bumped.)